src/Tutor.Infrastructure/Database/Repositories/Domain/AssessmentEventDatabaseRepository.cs

```diff
@@ -40,5 +40,16 @@ public void SaveSubmission(Submission submission)
             _dbContext.Submissions.Attach(submission);
             _dbContext.SaveChanges();
         }
+
+        public Submission FindSubmissionWithMaxCorrectness(Submission submission)
```

Comment:
Do we need the whole submission or just the AeId and LearnerId? In this case, the method becomes clearer when you work with these two params instead of a single object that has more information that is needed here.